components/infra-proxy-service/server/databags.go

```diff
@@ -159,6 +159,43 @@ func (s *Server) DeleteDataBag(ctx context.Context, req *request.DataBag) (*resp
 
 }
 
+// UpdateDataBagItem updates a data bag item
+func (s *Server) UpdateDataBagItem(ctx context.Context, req *request.UpdateDataBagItem) (*response.UpdateDataBagItem, error) {
+	c, err := s.createClient(ctx, req.OrgId, req.ServerId)
+	if err != nil {
+		return nil, err
+	}
+
+	if req.Name == "" {
+		return nil, status.Error(codes.InvalidArgument, "must supply data bag name")
+	}
+
+	if req.ItemId == "" {
+		return nil, status.Error(codes.InvalidArgument, "must supply data bag item name")
+	}
+
+	data := req.Data
+	if data == "" {
+		data = "{}"
+	}
+
+	var dataBagItem chef.DataBagItem
+	err = json.Unmarshal(json.RawMessage(data), &dataBagItem)
```

Comment:
See several comments on these various items in #3866 (validate.RequiredName, defaultIfEmpty, and structpb).